docs/vec_of_option.md

```diff
@@ -0,0 +1,266 @@
+
+# 12. What are your `Option`s?
```

Comment:
😄 nice

Reply:
an alternative title which comes to mind is "I got, I got, I got, I got options" 😄 https://youtu.be/xALCpLXcIYY?si=X7liC_24CV86p3yY

Reply:
I'm fine with both, but you'll have to be the one to pick! I didn't know either of these songs :sweat_smile:  
Maybe the most famous one? I would guess the first by the number of views